stable_baselines/gail/dataset/record_expert.py

```diff
@@ -101,19 +108,31 @@ def generate_expert_traj(model, save_path, env=None, n_timesteps=0,
             observations.append(obs)
 
         if isinstance(model, BaseRLModel):
-            action, _ = model.predict(obs)
+            action, state = model.predict(obs, state=state, mask=mask)
         else:
             action = model(obs)
 
         obs, reward, done, _ = env.step(action)
 
+        # Use only first env
+        if is_vec_env:
+            if isinstance(reward, float):
```

Comment:
In which case does that apply?
Even with only one VecEnv, I would expect a list, not a float, or maybe you wanted that to be outside `if is_vec_env` ?

Btw, you can simplify the code using:
```python
mask = np.array(done)
```
in the case of VecEnv

Reply:
Models with only single env support are only returning a float even if they are using a VecEnv. This crashed the first Travis test.

Reply:
I think this is true only for `stable_baselines.common.base_class._UnvecWrapper` which is used by DQN, SAC and DDPG. It is a wrapper not part of the public api but that is useful to make VecEnv work with all the algos.

So I think we should update the check `is_vec_env` to exclude that type of env...

See code below for normal VecEnv ;)
```python
import gym
import numpy as np

from stable_baselines.common.vec_env import DummyVecEnv

env = DummyVecEnv([lambda: gym.make("Pendulum-v0")])
env.reset()
_, _, done, _ = env.step([env.action_space.sample()])

# Env with a single env returns np.ndarray
assert not isinstance(done, float)
assert isinstance(done, np.ndarray)
```

Reply:
sorry haven't seen it, wrong notification setting.

I would be cautious about that, it could break user code when using `DQN`, `SAC` and `DDPG`. checking for a float each frame is of course also not the best solution, but it keeps supporting `stable_baselines.common.base_class._UnvecWrapper`

Reply:
well if you replace:
```python
if isinstance(env, VecEnv):
```
by
```python
if isinstance(env, VecEnv) and not isinstance(env, _UnvecWrapper):
```
that should be ok, no? (in the sense that you could remove your check)

Reply:
I misunderstand your Idea, this of course works.